src/fields/equation_output.hh

```diff
@@ -93,9 +97,29 @@ public:
      */
     void output(TimeStep step);
 
+    /// Getter for set of observe fields.
+    inline const std::unordered_set<string> &observe_fields() const {
```

Comment:
We should not pass the whole EquationOutput to the observe assembly and implement those getters, rather pass necessary data through the constructor, possibly wrapping them into an auxiliary struct.